flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/JsonDebeziumSchemaSerializer.java

```diff
@@ -113,6 +138,67 @@ public byte[] serialize(String record) throws IOException {
         return objectMapper.writeValueAsString(valueMap).getBytes(StandardCharsets.UTF_8);
     }
 
+    public boolean schemaChangeV2(JsonNode recordRoot) {
+        boolean status = false;
+        try {
+            if (!StringUtils.isNullOrWhitespaceOnly(sourceTableName) && !checkTable(recordRoot)) {
+                return false;
+            }
+            List<String> ddlSqlList = extractDDLList(recordRoot);
+            if (CollectionUtils.isEmpty(ddlSqlList)) {
+                LOG.info("ddl can not do schema change:{}", recordRoot);
+                return false;
+            }
+
+            List<DDLSchema> ddlSchemas = SchemaChangeHelper.getDdlSchemas();
+            for (int i = 0; i < ddlSqlList.size(); i++) {
+                DDLSchema ddlSchema = ddlSchemas.get(i);
+                String ddlSql = ddlSqlList.get(i);
+                boolean doSchemaChange = checkSchemaChange(ddlSchema);
+                status = doSchemaChange && execSchemaChange(ddlSql);
+                LOG.info("schema change status:{}", status);
+            }
+        } catch (Exception ex) {
+            LOG.warn("schema change error :", ex);
+        }
+        return status;
+    }
+
+    private boolean checkSchemaChange(DDLSchema ddlSchema) throws IOException, IllegalArgumentException {
+        String requestUrl = String.format(CHECK_SCHEMA_CHANGE_API, RestService.randomEndpoint(dorisOptions.getFenodes(), LOG), database, table);
+        Map<String,Object> param = buildRequestParam(ddlSchema);
+        HttpGetWithEntity httpGet = new HttpGetWithEntity(requestUrl);
+        httpGet.setHeader(HttpHeaders.AUTHORIZATION, authHeader());
+        httpGet.setEntity(new StringEntity(objectMapper.writeValueAsString(param)));
+        boolean success = handleResponse(httpGet);
+        if (!success) {
+            LOG.warn("schema change can not do table {}.{}",database,table);
+        }
+        return success;
+    }
+
+    private List<String> extractDDLList(JsonNode record) throws JsonProcessingException {
```

Comment:
Need to add unit tests

Reply:
Thanks for your suggestion, I have changed.